core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/ShaclSailConnection.java

```diff
@@ -386,14 +388,17 @@ private ValidationReport performValidation(List<Shape> shapes, boolean validateE
 		try {
 			Stream<Callable<ValidationResultIterator>> callableStream = shapes
 					.stream()
-					.map(shape -> shape.generatePlans(connectionsGroup, sail.isLogValidationPlans(),
-							validateEntireBaseSail))
-					.filter(Objects::nonNull)
-
-					.map(SingleCloseablePlanNode::new)
-
-					.map(planNode -> () -> {
-
+					.map(shape -> new ShapePlanNodeTuple(shape,
```

Comment:
ShapePlanNodeTuple allows us to re-enable some of the logging that was previously commented out.

Reply:
Eg. it gives us access to the underlying shape